jetty-websocket/websocket-core-common/src/main/java/org/eclipse/jetty/websocket/core/WebSocketComponents.java

```diff
@@ -43,41 +46,64 @@ public WebSocketComponents()
     public WebSocketComponents(WebSocketExtensionRegistry extensionRegistry, DecoratedObjectFactory objectFactory,
                                ByteBufferPool bufferPool, InflaterPool inflaterPool, DeflaterPool deflaterPool)
     {
-        this.extensionRegistry = (extensionRegistry == null) ? new WebSocketExtensionRegistry() : extensionRegistry;
-        this.objectFactory = (objectFactory == null) ? new DecoratedObjectFactory() : objectFactory;
-        this.bufferPool = (bufferPool == null) ? new MappedByteBufferPool() : bufferPool;
-        this.inflaterPool = (inflaterPool == null) ? new InflaterPool(CompressionPool.DEFAULT_CAPACITY, true) : inflaterPool;
-        this.deflaterPool = (deflaterPool == null) ? new DeflaterPool(CompressionPool.DEFAULT_CAPACITY, Deflater.DEFAULT_COMPRESSION, true) : deflaterPool;
+        this (extensionRegistry, objectFactory, bufferPool, inflaterPool, deflaterPool, null);
+    }
+
+    public WebSocketComponents(WebSocketExtensionRegistry extensionRegistry, DecoratedObjectFactory objectFactory,
+                               ByteBufferPool bufferPool, InflaterPool inflaterPool, DeflaterPool deflaterPool, Executor executor)
+    {
+        _extensionRegistry = (extensionRegistry == null) ? new WebSocketExtensionRegistry() : extensionRegistry;
+        _objectFactory = (objectFactory == null) ? new DecoratedObjectFactory() : objectFactory;
+        _bufferPool = (bufferPool == null) ? new MappedByteBufferPool() : bufferPool;
+        _inflaterPool = (inflaterPool == null) ? new InflaterPool(CompressionPool.DEFAULT_CAPACITY, true) : inflaterPool;
+        _deflaterPool = (deflaterPool == null) ? new DeflaterPool(CompressionPool.DEFAULT_CAPACITY, Deflater.DEFAULT_COMPRESSION, true) : deflaterPool;
 
-        addBean(inflaterPool);
-        addBean(deflaterPool);
-        addBean(bufferPool);
-        addBean(extensionRegistry);
-        addBean(objectFactory);
+        if (executor == null)
+        {
+            QueuedThreadPool threadPool = new QueuedThreadPool();
+            threadPool.setName("WebSocket@" + hashCode());
+            _executor = threadPool;
+        }
+        else
+        {
+            _executor = executor;
+        }
```

Comment:
If you are breaking out like this, then you may as well do:

```suggestion
        if (executor == null)
        {
            QueuedThreadPool threadPool = new QueuedThreadPool();
            threadPool.setName("WebSocket@" + hashCode());
            _executor = threadPool;
            addBean(_executor, true);
        }
        else
        {
            _executor = executor;
            addBean(_executor, false);
        }
```

Reply:
Hmmm but then all the other beans are managed differently.....  I still don't like how the management decision is made external to this class... but I'll live with it in this PR.   But do think about fixing it sometime.